zerver/signals.py

```diff
@@ -0,0 +1,66 @@
+from django.dispatch import receiver
+from django.contrib.auth.signals import user_logged_in
+from django.core.mail import send_mail
+from django.conf import settings
+from django.template import loader
+from django.utils import timezone
+from typing import Any, Dict
+from zerver.models import UserProfile
+
+
+def get_device_browser(user_agent):
+    # type: (str) -> str
+
+    if "chrome" in user_agent and "chromium" not in user_agent:
+        return 'Chrome'
+    elif "firefox" in user_agent and "seamonkey" not in user_agent and "chrome" not in user_agent:
+        return "Firefox"
+    elif "chromium" in user_agent:
+        return "Chromium"
+    elif "safari" in user_agent and "chrome" not in user_agent and "chromium" in user_agent:
```

Comment:
I did have bug in testing for Safari browser. As 'chrome' and  'chromium' should not be in the string.